pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstanceassembler/FormInstanceCreator.java

```diff
@@ -0,0 +1,106 @@
+package org.broadinstitute.ddp.service.actvityinstanceassembler;
+
+
+import org.broadinstitute.ddp.content.ContentStyle;
+import org.broadinstitute.ddp.content.HtmlConverter;
+import org.broadinstitute.ddp.db.dao.ActivityI18nDao;
+import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
+import org.broadinstitute.ddp.model.activity.definition.FormActivityDef;
+import org.broadinstitute.ddp.model.activity.definition.i18n.ActivityI18nDetail;
+import org.broadinstitute.ddp.model.activity.definition.template.Template;
+import org.broadinstitute.ddp.model.activity.instance.FormInstance;
+
+import java.util.Optional;
+
+/**
+ * Creates {@link FormInstance}
+ */
+public class FormInstanceCreator extends ElementCreator {
+
+    public FormInstanceCreator(ActivityInstanceAssembleService.Context context) {
+        super(context);
+    }
+
+    public FormInstance createFormInstance() {
+        ActivityI18nDetail activityI18nDetail = getActivityI18nDetail();
+        FormInstance formInstance = constructFormInstance(activityI18nDetail);
+        addChildren(formInstance);
+        updateBlockStatuses(formInstance);
+        render(formInstance);
+        return formInstance;
+    }
+
+    private FormInstance constructFormInstance(ActivityI18nDetail activityI18nDetail) {
+        ActivityInstanceDto instanceDto = context.getActivityInstanceDto();
+        FormActivityDef formActivityDef = context.getFormActivityDef();
+
+        FormInstance formInstance = new FormInstance(
+                instanceDto.getParticipantId(),
+                instanceDto.getId(),
+                instanceDto.getActivityId(),
+                instanceDto.getActivityCode(),
+                formActivityDef.getFormType(),
+                instanceDto.getGuid(),
+                activityI18nDetail != null ? activityI18nDetail.getTitle() : null,
```

Comment:
The `FormActivityDef` object should already have things like title and subtitle. I think we should reuse those instead of querying for an `ActivityI18nDetail`. Take a look at `FormActivityDao::findDefByDtoAndVersion()` method -- it already queries for those `ActivityI18nDetail`. I think what you need to do here is iterate through those lists and pick out the right one using the language code.